src/util/cron.c

```diff
@@ -283,9 +288,11 @@ void Cron_AbortTask
 		return;
 	}
 	
+	pthread_mutex_lock(&cron->mutex);
```

Comment:
it seems like all calls to `CRON_TaskAdvanceState` are governed by `pthread_mutex_lock` and `pthread_mutex_unlock`
please update `CRON_TaskAdvanceState` to acquire and release the mutex within the function itself.